drivers/storage/portworx/portworx.go

```diff
@@ -149,16 +150,16 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1alpha1.StorageClu
 
 	toUpdate.Spec.Image = strings.TrimSpace(toUpdate.Spec.Image)
 	toUpdate.Spec.Version = pxutil.GetImageTag(toUpdate.Spec.Image)
-	pxVersionChanged := toUpdate.Spec.Version == "" ||
-		toUpdate.Spec.Version != toUpdate.Status.Version
+	pxVersionChanged := pxEnabled &&
+		(toUpdate.Spec.Version == "" || toUpdate.Spec.Version != toUpdate.Status.Version)
```

Comment:
what if `toUpdate.Status.Version` is already `""` ? Is an update required in this case?

Reply:
Yes. That will happen only the first time this version of operator is started (spec.Version present but status.Version absent), because the status.Version field was newly introduced. The status.Version will be set in the first reconcile loop and after that it  won't require the update until spec.Version changes.